manifests/backend.pp

```diff
@@ -176,6 +179,7 @@
   Stdlib::Port $postgresql_port = 5432,
   String $postgresql_dbname = 'sensu',
   Integer $postgresql_pool_size = 20,
+  Boolean $declare_cli_class = true,
```

Comment:
This is a new patter, why should the user care how the class is included?

Reply:
Because if someone wants to do this:

```puppet
class { '::sensu::cli': ... }
class { '::sensu::backend': ... }
```

Then they will duplicate resource errors unless they set `declare_cli_class => false`.

Reply:
Could we use data in module to work around this?

Reply:
I don't know how that would work? This is something I think we just have to allow someone to turn on/off based on how they use this module.

Reply:
I think only way to avoid this would be to do something like `if ! defined(Class['sensu::cli'])` but I thought such order dependent hacks were discouraged in Puppet.

Reply:
Why do we have to use `class` at all instead of using`include`?

Reply:
Because then instead of this:

```
class { 'sensu::backend': url_host => 'foo' }
```

A site has to do this with just include: 

```
class { 'sensu::cli': url_host => 'foo' }
class { 'sensu::backend': url_host => 'foo' }
```

There are parameters that we pass to backend class that are also needed by CLI class and this maintains backwards compatibility for behavior too.

Reply:
One idea is to move `url_host` and `url_port` to `sensu` class since shared between backend agent and now CLI. This would be easier once #1159 is merged and #1159 will have to be updated once this PR is merged. If we move these and other parameters to `sensu` class we could simplify with `include` but would be breaking change. I can make that change as part of #1159

Reply:
I like that approach. Let's do this as part of the breaking change.

Reply:
So implement that breaking change in this PR or keep this current behavior and do new PR for breaking change?

Reply:
This was implemented.